openthread/Kconfig

```diff
@@ -24,4 +26,82 @@ config OPENTHREAD_LIBRARY_1_1
 
 endchoice
 
+# OpenThread configuration features sets from Nordic Semiconductor
+
+choice
+	prompt "Nordic library features set"
```

Comment:
Current 
--------

This looks as:
![image](https://user-images.githubusercontent.com/9802583/91813457-046d9880-ec33-11ea-85b4-45995d2a7357.png)
and
![image](https://user-images.githubusercontent.com/9802583/91813502-16e7d200-ec33-11ea-869c-4d5c1e4dee85.png)

Suggestion
-------

How about making it a config, and when enabled, then choose the configuration, like this instead:
![image](https://user-images.githubusercontent.com/9802583/91814144-fc622880-ec33-11ea-8729-c41ae6df2bee.png)

and when selected:
![image](https://user-images.githubusercontent.com/9802583/91814223-156ad980-ec34-11ea-8490-7f899c97bd2d.png)

and selection of configuration:
![image](https://user-images.githubusercontent.com/9802583/91814292-303d4e00-ec34-11ea-8443-77b630efb7a6.png)

That way there is no need for a custom, and it is very easy to see that when a Nordic configuration is used.

Diff: 
```
index 1e5c309..4f09ee2 100644
--- a/openthread/Kconfig
+++ b/openthread/Kconfig
@@ -28,17 +28,17 @@ endchoice
 
 # OpenThread configuration features sets from Nordic Semiconductor
 
+config OPENTHREAD_NORDIC_LIBRARY
+       bool
+       prompt "Nordic library feature sets"
+
 choice
-       prompt "Nordic library features set"
-       default OPENTHREAD_NORDIC_LIBRARY_CUSTOM
+       prompt "Feature set"
+       default OPENTHREAD_NORDIC_LIBRARY_MASTER
+       depends on OPENTHREAD_NORDIC_LIBRARY
        help
          Nordic Semiconductor features set selection for OpenThread stack.
 
-config OPENTHREAD_NORDIC_LIBRARY_CUSTOM
-       bool "Custom OpenThread configuration"
-       help
-         Custom configuration of OpenThread features.
-
 config OPENTHREAD_NORDIC_LIBRARY_MASTER
        bool "Nordic OpenThread master configuration"
        select OPENTHREAD_BORDER_AGENT
```
```

Reply:
Yes, good idea.

Reply:
I reverted these changes since they forced to set `OPENTHREAD_NORDIC_LIBRARY=y` in the projects.

Setting `OPENTHREAD_NORDIC_LIBRARY_MASTER=y` alone would result in a warning and none of that configuration being built.

Reply:
Not sure I understand that argument.
Now users can do:
```
OPENTHREAD_NORDIC_LIBRARY=y
```
and then `OPENTHREAD_NORDIC_LIBRARY_MASTER=y` will be set per default, because of:
```
choice
       prompt "Feature set"
       default OPENTHREAD_NORDIC_LIBRARY_MASTER
```

Reply:
Yes, but that doesn't seem clean to me, normally it would require two settings:

```
OPENTHREAD_NORDIC_LIBRARY=y
OPENTHREAD_NORDIC_LIBRARY_FTD=y
```

Is this acceptable?

Reply:
> Is this acceptable?

I would say so.
That is aligned with, for example Bluetooth:
```
CONFIG_BT=y
CONFIG_BT_CENTRAL=y
```
although Bluetooth does default to Nordic softdevice, when using NCS, so that `BT_LL_SOFTDEVICE` is not required above.
```
choice BT_LL_CHOICE
...
	default BT_LL_SOFTDEVICE
```
https://github.com/nrfconnect/sdk-nrfxlib/blob/master/softdevice_controller/Kconfig#L29

It could also make sense to have
```
config OPENTHREAD_NORDIC_LIBRARY
       bool
       prompt "Nordic library feature sets"
       default y
```
but I didn't propose it initially, as I don't know if you default prefer users to use custom or a nordic reference conf when using NCS.

That way, `OPENTHREAD_NORDIC_LIBRARY_FTD=y` would be sufficient in a conf file.
But I see no problem in specifying both conf settings.

Reply:
OK, reverting back the changes and updating the master PR.